aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/CreateHandler.java

```diff
@@ -0,0 +1,54 @@
+package software.amazon.rds.optiongroup;
+
+import java.util.Optional;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.OptionGroupAlreadyExistsException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    public static final int PHYSICAL_RESOURCE_ID_MAX_LENGTH = 255;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<RdsClient> proxyClient,
+        final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+        if (StringUtils.isNullOrEmpty(model.getOptionGroupName())) {
+            model.setOptionGroupName(IdentifierUtils.generateResourceIdentifier(
+                    Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse("optiongroup"),
+                    request.getClientRequestToken(),
+                    PHYSICAL_RESOURCE_ID_MAX_LENGTH
+            ).toLowerCase());
+        }
+
+        return proxy.initiate("rds::create-option-group", proxyClient, model, callbackContext)
+                .translateToServiceRequest(resourceModel -> {
+                    return Translator.createOptionGroupRequest(model, request.getDesiredResourceTags());
+                })
+                .backoffDelay(CONSTANT)
+                .makeServiceCall((createOptionGroupRequest, proxyInvocation) -> {
+                    return proxyInvocation.injectCredentialsAndInvokeV2(createOptionGroupRequest, proxyInvocation.client()::createOptionGroup);
+                })
+                .handleError((createOptionGroupRequest, exception, client, resourceModel, ctx) -> {
+                    if (exception instanceof OptionGroupAlreadyExistsException) {
+                        return ProgressEvent.defaultFailureHandler(exception, HandlerErrorCode.AlreadyExists);
+                    }
+                    throw exception;
+                })
+                .progress()
+                .then(progress -> updateOptionGroup(proxy, proxyClient, progress))
```

Comment:
can some properties not be applied in the createOptionGroupRequest?

Reply:
Surprisingly, but yes:
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateOptionGroup.html
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyOptionGroup.html
It seems one can't specify Options while creating an object: only upon update.